Javascript2 week2/dennis

DennisAgyekum opened the pull request.

DennisAgyekum committed:
Added a toble to display currency rate

DennisAgyekum committed:
chnaged the whole css styling

DennisAgyekum committed:
Implement a function to find a specific currency rate, searching by the currency from or to.

vdaguenet commented on currency-converter/currency.css:
```diff
@@ -65,25 +112,259 @@ input[type="number"] {
+  z-index: 1;
+}
+
+.squares {
```
I would name this class `.flags` because that it's what it contains.

DennisAgyekum replied:
i have renamed the class to flags

vdaguenet commented on currency-converter/currency.html:
```diff
@@ -47,13 +77,49 @@ <h2>Update Existing Currency Rate</h2>
+          </table>
+        </div>
+      </section>
+      <script src="currency.js"></script>
```
Always keep your `script` tags at the end of the `<body>`.

DennisAgyekum replied:
script has been moved to the end of the body

vdaguenet commented on currency-converter/currency.js:
```diff
@@ -1,80 +1,163 @@
+        .trim();
+      const rate = parseFloat(document.getElementById("updateRate").value);
+
+      const rateObj = currencyRates.find((rate) => rate.base === baseCurrency);
```
This is a good start for finding a rate when the `base` and `to` currencies are provided. However, the code needs to also work when only one currency is provided. Have a look at my messages on Slack. You can either make a search function with two arguments like [this](https://hackyourfuture-cph.slack.com/archives/C07AT59R9QR/p1720790626662359) or have only one argument like [this](https://hackyourfuture-cph.slack.com/archives/C07AT59R9QR/p1720794065032809).

DennisAgyekum replied:
i have changed the search input to display the available rates even if one of the fields id empty.

vdaguenet commented on currency-converter/currency.js:
```diff
@@ -1,80 +1,163 @@
-      USD: 1.23,
-      GBP: 0.88,
-      EUR: 1.82,
+  const currencyRates = [
```
Well done! This part was not super clear from the brief but you understood it well 👍🏻

DennisAgyekum committed:
Update currency.html

moved the script element to end of the body

DennisAgyekum committed:
Update currency.html

changed the class name to flags

DennisAgyekum committed:
Update currency.css

DennisAgyekum committed:
Update currency.js

changed the search form to display available currency rates even if one input is empty

DennisAgyekum committed:
Update currency.html

removed to required attribute from the html.

DennisAgyekum committed:
Update currency.html

vdaguenet reviewed (approved):
Nice!
